File: java/fileOperations.java
import java.io.*;
public class fileOperations {

   static void readOne() throws IOException {
      FileInputStream in = new FileInputStream("../example.txt");
      if (in != null) {
         in.close();
      }

      int c;
      while ((c = in.read()) != -1) {
         System.out.println(c);
      }
   }

   static void readTwo() throws IOException {
      FileInputStream in = null;

      try {
         in = new FileInputStream("../example.txt");
         
         int c;
         while ((c = in.read()) != -1) {
            System.out.println(c);
         }
      }finally {
         if (in != null) {
            in.close();
            in.close();
         }
      }
   }

   static void readThree() throws IOException {
      FileInputStream in = null;

      in = new FileInputStream("../example.txt");
      
      int c;
      while ((c = in.read()) != -1) {
         System.out.println(c);
      }  
   }

   public static void main(String args[]) throws IOException {  
      readTwo();
      readOne();
      readThree();
   }
}

File: python/fileOperations.py
def read_one():
	fo = open("example.txt", "r")
	fo.write("something")

def read_two():
	fo = open("example.txt", "r")
	print(fo.read())
	fo.close()
	fo.close()

def read_three():
	fo = open("example.txt", "r")
	fo.close()
	print(fo.read())

def read_four():
	fo = open("example.txt", "r")
	print(fo.read())

if __name__ == '__main__':
	read_two()
	read_three()
	read_four()
	read_one()